File: src/main/java/Start.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;

public class Start {

    public Start() {
        // ----------------------------------------------
        int problemNumber = 1;
        // ----------------------------------------------

        try {
            StringBuilder pN = new StringBuilder("" + problemNumber);
            while (pN.length() < 4) pN.insert(0, "0");

            System.out.println("\033[4mProblem " + problemNumber + "\n\033[0m");

            // TODO: GET DESCRIPTION FROM WEB
            try {
                BufferedReader br = new BufferedReader(new FileReader(new File("src/problem/description/problem_" + pN + ".txt")));
                String line;
                while ((line = br.readLine()) != null) System.out.println(line);
                br.close();
            } catch (FileNotFoundException fnfe) {
                System.out.println("\033[91mNo problem description found!\033[0m");
            }
            // ---------

            System.out.println();

            //TODO: add dynamic package
            String pack = "p0" + problemNumber / 100;
            long start_time = System.nanoTime();
            Object solution =
                    Class.forName(pack + ".Problem_" + pN).getDeclaredConstructor().newInstance().getClass().getMethod("solve").invoke(null);
            long end_time = System.nanoTime() - start_time - 300000;

            System.out.println("Solution: " + solution);
            System.out.println("Time:     " + timeFormat(end_time));
            System.out.println("         [ s| ms| μs| ns]");

        } catch (Exception e) {
            System.err.println("Problem not solved yet!");
            e.printStackTrace();
        }
    }

    public static void main(String[] args) {
        new Start();
    }

    private String timeFormat(long l) {
        StringBuilder t = new StringBuilder(l + "");

        while (t.length() < 11) t.insert(0, 0);

        for (int i = 8; i > 0; i -= 3)
            t = new StringBuilder(t.substring(0, i) + ":" + t.substring(i));

        return t.toString();
    }

}

File: src/main/python/start.py
from importlib import import_module
from time import time_ns as time

import requests
from bs4 import BeautifulSoup

if __name__ == '__main__':

    # ----------------------------------------------
    problem_number = 1
    # ----------------------------------------------

    # TODO: add MathJax parsing
    print('\033[4mProblem %s\n\033[0m' % problem_number)
    try:
        print("\n".join([s.strip() for s in
                         BeautifulSoup(requests.get("https://projecteuler.net/problem=" + str(problem_number)).content, "html.parser").find_all("div", {"class": "problem_content"})[0].get_text()
                        .replace("\n", " ").replace(". ", ".\n").split("\n")]))
    except IndexError:
        print('\033[91mNo problem description found!\033[0m')

    try:
        package = str(problem_number // 100)
        problem = import_module('python.p%s.p_%s' % ('0' * (2 - len(package)) + package, '0' * (4 - len(str(problem_number))) + str(problem_number)))

        start_time = time()
        sol = problem.solve()
        total_time = time() - start_time

        t = '0' * (11 - len(str(total_time))) + str(total_time)
        for i in [-9, -6, -3]:
            t = t[:i] + ':' + t[i:]

        print('Solution:', sol)
        print('Time:    ', t)
        print('         [ s| ms| μs| ns]')

    except ModuleNotFoundError:
        print('\033[91mProblem not solved yet!\033[0m')
